### PR TITLE
feat: add a new action to post release information on comments

### LINKED ISSUE #12066: [Feature Request]: Add a bot to comment on PRs/issues when they are released that includes the version to pick up
### Summary

Add a bot or some sort of standard for adding a comment to PRs/issues when the bug/feature is released, including the version that the change is picked up in.

### Justification

When I’m trying to fix a bug with carbon, I often find that it is already an issue that was fixed and released, but then I have to search through pages of release notes and search for the exact issue/PR number to find what version the change was picked up in.

### Screenshots

Currently to find the release that a PR/issue is included in you have to do some digging, and it's not easily accessible for developers. 

### Desired UX and success metrics

_No response_

### Required functionality

_No response_

### Specific timeline issues / requests

_No response_

### Your team

Cloud Pak for Data

### Available extra resources

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Shankar-CodeJunkie**
Closes #12066

I've created a action, which compares the 2 latest tags/revisions. The comparison gives a list of commits, and then we will use ` GET /repos/${owner}/${repo}/commits/${commit}/pulls` to identify the pull request associated for that commit.  We will then post a message to that pull request. 
The action will run after every successful new release published on the github. 

#### Changelog

**New**

This PR will add a comment with release information on all the new pull request after every release. 

#### Testing / Reviewing

image

### CODE REVIEW

**Comment by Shankar-CodeJunkie** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by Shankar-CodeJunkie** [comment]
recheck

**Comment by tay1orjones** [comment]
Status checks were failing - I updated the yarn lockfile/cache and pushed it to this PR.

**Comment by ShankarVenkatachalam-official** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by ShankarVenkatachalam-official** [comment]
recheck

File: .github/workflows/release-notifications.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Release Notifications
+
+on:
+  release:
+    types: [published]
```
Comment (tay1orjones) [inline .github/workflows/release-notifications.yml]: I don't think we'll want to run this for all our prereleases, only on stable releases. [Docs reference](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-42)

```suggestion
    types: [released]
```

**Review by tay1orjones — APPROVED** [review]
Thanks so much for this submission! This will be a very welcome improvement to knowing the release that a PR was initially included in.

I hope you don't mind, I took the liberty of making a few minor adjustments:
- Renamed the action to `release-notifications`
- Improved logging throughout the action
- Updated the token to use our `carbon-bot` account token

**Review by tw15egan — APPROVED** [review]
This is awesome! Thanks for contributing this 🎉 👍🏻 ✅

File: actions/release-notifications/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+const core = require('@actions/core');
```
Comment (tay1orjones) [inline actions/release-notifications/index.js:8]: One thing we need here actually is the copyright header added to the file

**Review by tay1orjones — CHANGES_REQUESTED** [review]

File: actions/release-notifications/index.js (hunk 2 above)
Comment (Shankar-CodeJunkie) [reply actions/release-notifications/index.js:8]: Hi @tay1orjones : Yes. I've added it now. Please let me know in case of any changes needed. Thanks

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**